### PR TITLE
Updating templates for recent binding changes

### PR DESCRIPTION
**Pull request by mathewc**
Absorbing changes from https://github.com/Azure/azure-webjobs-sdk-script/commit/81ee8b71b4f6c807ee2cbb0661685114fcd81a70

Also making a few additional cleanups/fixes. E.g., I'm adding some missing samples to the sln file. I added the sln as a way for us to have a project to open to facilitate working on samples. I realize some people use Sublime, but I don't, so keeping this in sync helps me alot. What are others using? Either we keep the sln in sync or remove it - I'm for keeping it :)

### CODE REVIEW

File: Templates/HttpTrigger/index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ﻿module.exports = function (context, req) {
-    context.log('Node.js HTTP trigger function processed a request. Name=' + req.originalUrl);
+    context.log('Node.js HTTP trigger function processed a request. RequestUri=' + req.originalUrl);
```
Comment (mathewc) [inline Templates/HttpTrigger/index.js]: Fix for https://github.com/Azure/azure-webjobs-sdk-templates/issues/31

**Comment by davidebbo** [comment]
:shipit: But be aware that it goes live as soon as you merge. So if it's dependent on your script changes, let's not merge till those are live.

File: Templates/QueueTrigger-Python/function.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
       },
       {
         "type": "table",
```
Comment (mathewc) [inline Templates/QueueTrigger-Python/function.json]: All of the json files have an extra level of unecessary indenting. Anyone know why, or whose editor is doing this?

Comment (fabiocav) [inline Templates/QueueTrigger-Python/function.json]: I believe @soninaren is using VS Code with the default indentation configuration (4 spaces) and that has updated our JSON files.

**Comment by mathewc** [comment]
I'll hold off committing this until we update, since as David says these updates rely on runtime updates.

File: Templates/QueueTrigger-Python/function.json (hunk 2 above)
Comment (mathewc) [inline Templates/QueueTrigger-Python/function.json]: So what is everyone else using, 4 spaces? Both my machines are using 2 (VS Enterprise) and I don't recall changing any defaults.

Comment (davidebbo) [inline Templates/QueueTrigger-Python/function.json]: VS uses 4 spaces for me for JSON. Not sure what;s the default, but we should all use the same...

**Comment by soninaren** [comment]
this has been merged in already.

**Pull request closed without merging**